src/Dapr.Client/DaprClientGrpc.cs

```diff
@@ -1373,6 +1375,475 @@ private Autogenerated.StateItem ToAutogeneratedStateItem(StateTransactionRequest
             var resp = await client.UnsubscribeConfigurationAsync(request, options);
             return new UnsubscribeConfigurationResponse(resp.Ok, resp.Message);
         }
+
+        #endregion
+
+        #region Cryptography
+
+        private async Task<byte[]> EncryptStreamAsync(string vaultResourceName, Stream stream,
```

Comment:
Ideally in this method you would want to return a stream too, or the response is still buffered in-memory.

At least there should be a variant of this method that returns a stream so it can be written to file (or somewhere else) without buffering

Reply:
@philliphoff Mentioned this in another review note earlier, but the idea here was based on our conversation in Discord where you called out that the gRPC implementation doesn't support unbounded streams. Given that there's a specific length to the data, while the sidecar can work on this in chunks, I was concerned that by exposing a Stream response method, there might be developers that misunderstand that to create perpetual streams. By instead accepting a Stream input and returning a (now `ReadOnlyMemory<byte>`), it makes clear that there's a boundary to the data given and received and the SDK handles buffering the response to take that off the developer's plate.

I'm not opposed to adding an overload that returns a stream, but that was my reasoning in the first pass. Thoughts?

Reply:
I'm not sure I understand what you mean with the concerns about unbounded streams.

The API is designed to use bi-di streams just as a way to be able to get back the data that is encrypted (or decrypted) as soon as it's processed. In the Go and JS implementations, we are able to encrypt files that are multiple GBs in size without needing to buffer more than a few KBs in memory at any time.

The idea would be to return a readable stream to the caller. I am not sure how that is implemented in C#, but the caller would receive a stream and can attach a reader to that to consume the data as it comes in.

The non-streaming implementation would just be a wrapper around the method above, which buffers the data in memory and returns a byte array.

Reply:
Sure, all that makes sense to me. I've got no problem with adding another method to return a streamed response, but I fear it's going to get messy. Try as I might, I couldn't find a way to use the generated prototypes to make a generic implementation, so we're going to instead get several duplicated private methods for each. I'll add another commit in the next couple of hours.

Reply:
This now has encrypt and decrypt methods that return both `ReadOnlyMemory<byte>` and `IEnumerableAsync<byte[]>`. I refactored the implementations for each to minimize any duplicate code, so this should be in a better place now for byte-sized amounts of data or large streams.